### ISSUE TITLE
[Feature Request] Add rename option to hero selection screen

### ISSUE THREAD

**Issue by KornelJahn** [diasurgical/devilutionX#5401]
On the hero selection screen, it would be nice to also have a "Rename" option in addition to "OK", "Delete", and "Cancel".

![hero-rename-mockup](https://user-images.githubusercontent.com/538206/194712973-dd46e722-807a-426a-baec-21ae667725b3.png)

**Comment by qndel** [comment]
I really don't like that idea, people can enter your game, troll/ruin your game and then just come back under a different name and you will never know ...

**Comment by KornelJahn** [comment]
> I really don't like this idea, people can enter your game, troll/ruin your game and then just come back under a different name and you will never know ...

I agree with you that in-game renaming would make this kind of trolling easier but not enabling this feature would not prevent it from happening either. I mean, a troll could just as well use a hero editor. Even if a compatible hero editor does not exist now, it is just a matter of time until someone comes up with one. Moreover, if ["proposal: change save archive format from MPQ to JSON"](https://github.com/diasurgical/devilutionX/issues/614) got accepted and implemented in the future, you could rename a hero by simple text editing.

**Comment by qndel** [comment]
Yes but using a hero editor is much more effort, you can never stop trolls/cheaters but if you make trolling/cheating hard enough, there won't be many of these. If you can rename your character with a few clicks, more people will use it.

**Comment by AJenbo** [comment]
Since the saves are copatible with the original there are compatible editors.
